wowchemy/assets/js/wowchemy-slides.js

```diff
@@ -0,0 +1,188 @@
+}
+
+// chalkboard enabled by default
+if(pluginOptions.chalkboard_enabled === undefined){
+  pluginOptions.chalkboard_enabled = true
+}
+
+
+// configure chalkboard if enabled
+if (pluginOptions.chalkboard_enabled) {
+  enabledPlugins.push(RevealChalkboard);
+
+  // move chalkboard slightly to right if menu enabled
+  if (pluginOptions.menu_enabled) {
+    // make sure we have a config object
+    pluginOptions.chalkboard = (typeof pluginOptions.chalkboard === 'undefined') ? {} : pluginOptions.chalkboard;
+    pluginOptions.chalkboard.toggleChalkboardButton = { left: "80px" }
+    pluginOptions.chalkboard.toggleNotesButton = { left: "130px" }
+    
+  }
+}
+
+pluginOptions["plugins"] = enabledPlugins;
+
+Reveal.initialize(pluginOptions);
+
+
+// mermaid not enabled by default
+if(params.slides.mermaid_enabled === undefined){
+  params.slides.mermaid_enabled = false
+}
+
+// configure mermaid only if enabled
+if (params.slides.mermaid_enabled) {
+  //mermaid options
+  // mermaid: front matter configuration can be used to set mermaid options
+  // You can also use directives (see mermaid documentation)
+  var mermaidOptions = (typeof params.slides.mermaid_options === 'undefined') ? {} : params.slides.mermaid_options;
+
+  // startOnLoad must be false since diagrams are lazily rendered
+  mermaidOptions["startOnLoad"] = false;
+
+
+  mermaid.initialize(mermaidOptions);
+
+  // Following functions are used to render mermaid diagrams
+  // after reveal slides have been successfully loaded
+  // since content of slides is lazy loaded, if diagrams are
+  // rendered at start of presentation their sizes will be off
+  /* get all slides that are:
+1- data loaded
+2- display set to block
+3- has a mermaid element that is not processed (data-processed dne)
+
+*/
+  function mermaidSlidesReadyToRender(s) {
+    diag = s.querySelector(".mermaid");
+    if (diag) {
+      background = s.slideBackgroundElement;
+      // render if we are 1 slide away horizontally
+      // current visible slide index
+      chidx = Reveal.getState()["indexh"];
+
+      // mermaid slide index
+      shidx = Reveal.getIndices((slide = s))["h"];
+      if (
+        !diag.hasAttribute("data-processed") &&
+        background.hasAttribute("data-loaded") &&
+        background.style.display === "block" &&
+        shidx - chidx <= 1
+      )
+        return s;
+    }
+    return null;
+  }
+
+  function renderMermaidSlides() {
+    dSlides = Reveal.getSlides().filter(mermaidSlidesReadyToRender);
+    dSlides.forEach((item) => mermaid.init(item.querySelector(".mermaid")));
+  }
+
+  // render mermaid slides for slides that are ready
+  Reveal.on("slidechanged", (event) => {
+    renderMermaidSlides();
+  });
+
+  // render mermaid slides for slides that are ready on startup
+  Reveal.on("Ready", (event) => {
+    if (Reveal.isReady()) {
+      renderMermaidSlides();
+    }
+  });
+  // end of mermaid diagram functions
+
+  function fixMermaid() {
```

Comment:
Did you duplicate this `fixMermaid` code from the main Wowchemy JS file?

Reply:
I like live dangerously .. any other suggestions? its just a single function I think.

It fixed the issue where the \```mermaid directive doesnt work

Reply:
`fixMermaid()` has been refactored to a dedicated JS module `wowchemy-utils.js` now so that that specific function can be imported into this PR. Also, the function now uses vanilla JS rather than JQuery.

Reply:
@gcushen ok I have done what I can

Last change should address the last of your comments

if merging please take care of this part to use the Wowchemy-utils.js

I dont see it merged to master so I cannot use it. I need to also shift my attention somewhere else

thanks for the feedback

Reply:
> if merging please take care of this part to use the Wowchemy-utils.js
> I dont see it merged to master so I cannot use it. I need to also shift my attention somewhere else

`wowchemy-utils.js` is located here: https://github.com/wowchemy/wowchemy-hugo-modules/blob/master/wowchemy/assets/js/wowchemy-utils.js